h2o-core/src/main/java/water/util/CompressionFactory.java

```diff
@@ -0,0 +1,57 @@
+package water.util;
+
+import water.Iced;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.reflect.Constructor;
+import java.util.zip.GZIPOutputStream;
+
+public class CompressionFactory extends Iced<CompressionFactory> {
+
+  private String _name;
+
+  private CompressionFactory(String name) {
+    _name = name;
+  }
+
+  OutputStream wrapOutputStream(OutputStream os) throws IOException {
```

Comment:
This method could perfectly be static, with an enum of the compression algorithm. No need to instantiate.

Reply:
It takes more than just an enum ;)